src/components/Widget/index.js

```diff
@@ -391,7 +394,33 @@ class Widget extends Component {
               dispatch(emitUserMessage(message));
             }
           }
+
+          // Let's drop all the messages that we had
+          dispatch(dropMessages());
+
+          // Reverse the messages to be ordered from latest to earliest message
+          messages.reverse();
```

Comment:
Move this inside the `if` statement in line 405 and update the condition to `if (messages && messages.length > 0)`